gvr-balloons/app/src/main/java/org/gearvrf/balloons/BalloonMain.java

```diff
@@ -0,0 +1,299 @@
+        };
+        mParticleSystem = new ParticleEmitter(gvrContext, mScene, particleCreator);
+        mParticleSystem.MaxDistance = 10.0f;
+        mParticleSystem.TotalParticles = 20;
+        mParticleSystem.EmissionRate = 3;
+        mParticleSystem.MinVelocity = 2;
+        mParticleSystem.MaxVelocity = 6;
+        mParticleSystem.setEmitterArea(new Vector3f(-5.0f, -3.0f, -2.0f), new Vector3f(5.0f, -3.01f, -5.0f));
+        particleRoot.attachComponent(mParticleSystem);
+        mScene.addSceneObject(particleRoot);
+        mParticleSystem.start();
+        mTimer = new Timer();
+        TimerTask gameOver = new TimerTask()
+        {
+            public void run() { gameOver(); }
+        };
+        long oneMinute = 60 * 1000;
+        mTimer.schedule(gameOver, oneMinute);
+    }
+
+    /*
+     * Make an array of materials for the particles
+     * so they will not all be the same.
+     */
+    ArrayList<GVRMaterial> makeMaterials(GVRContext ctx)
+    {
+        float[][] colors = new float[][] {
+                { 1.0f,   0.0f,   0.0f,   0.8f },
+                { 0.0f,   1.0f,   0.0f,   0.8f },
+                { 0.0f,   0.0f,   1.0f,   0.8f },
+                { 1.0f,   0.0f,   1.0f,   0.8f },
+                { 1.0f,   1.0f,   0.0f,   0.8f },
+                { 0.0f,   1.0f,   1.0f,   0.8f }
+        };
+        ArrayList<GVRMaterial> materials = new ArrayList<GVRMaterial>();
+        for (int i = 0; i < 6; ++i)
+        {
+            GVRMaterial mtl = new GVRMaterial(ctx);
+            mtl.setDiffuseColor(colors[i][0], colors[i][1], colors[i][2], colors[i][3]);
+            materials.add(mtl);
+        }
+        return materials;
+    }
+
+    /*
+     * Make a large sphere enclosing the scene with a pretty countryside.
+     */
+    GVRSceneObject makeBackground(GVRContext ctx)
+    {
+        Future<GVRTexture> tex = ctx.loadFutureCubemapTexture(new GVRAndroidResource(mGVRContext, R.raw.lycksele3));
+        GVRMaterial mtl = new GVRMaterial(ctx, GVRMaterial.GVRShaderType.Cubemap.ID);
+        mtl.setMainTexture(tex);
+        GVRSphereSceneObject environment = new GVRSphereSceneObject(ctx, 18, 36, false, mtl, 4, 4);
+        environment.getTransform().setScale(20.0f, 20.0f, 20.0f);
+
+        GVRDirectLight sunLight = new GVRDirectLight(mGVRContext);
+        sunLight.setAmbientIntensity(0.4f, 0.4f, 0.4f, 1.0f);
+        sunLight.setDiffuseIntensity(0.6f, 0.6f, 0.6f, 1.0f);
+        environment.attachComponent(sunLight);
+        return environment;
+    }
+
+    GVRTextViewSceneObject makeScoreboard(GVRContext ctx)
+    {
+        GVRTextViewSceneObject scoreBoard = new GVRTextViewSceneObject(ctx, 3, 2, "Score: 0");
+        GVRRenderData rdata = scoreBoard.getRenderData();
+        scoreBoard.getTransform().setPosition(-1.2f, 1.2f, -3.0f);
+        scoreBoard.setTextColor(Color.YELLOW);
+        //rdata.setAlphaBlend(true);
+        rdata.setAlphaToCoverage(true);
```

Comment:
Since you've discovered this last night, should there be a rdata.setDepthTest(false)? I find the score looking better with renderingOrder of OVERLAY.